app/models/event/reminder_scheduler.rb

```diff
@@ -0,0 +1,36 @@
+module Event::ReminderScheduler
+  extend ActiveSupport::Concern
+
+  REMINDER_CADENCE = {
+    first_reminder: 15.minutes,
+    second_reminder: 10.minutes,
+    last_reminder: 0.minutes
+  }.freeze
+
+  included do
+    after_save_commit :schedule_reminder
+  end
+
+  private
+
+  def schedule_reminder
```

Comment:
I suggest changing the approach. It's going to be cumbersome to update/remove jobs if the cadence or the event's schedule changes. You can use SolidQueue's [recurring tasks mechanism](https://github.com/rails/solid_queue/?tab=readme-ov-file#recurring-tasks) to enqueue a job that will generate the appropriate notifications.